feat(ops): SplitBySeparators — regex-only splitter (Rust+Py)

MrAnayDongre opened the pull request:
Adds `SplitBySeparators` (regex-only, no chunk-size planning).
Rust executor + registration; Python spec; schema matches `SplitRecursively`.
143 passed / 1 skipped (DB). Closes #1003. cc @georgeh0

georgeh0 commented on src/ops/functions/split_by_separators.rs:
```diff
@@ -0,0 +1,432 @@
+
+        // If no separators were provided, emit whole text once.
+        if self.regex.is_none() {
+            return emit_single_chunk_kv(full_text, self.spec.trim, self.spec.include_empty);
```
This function has ~40 lines of code, but I feel we don't need it.

We can reuse the logic below. Only the way to compute `ranges` are different.

georgeh0 commented on src/ops/functions/split_by_separators.rs:
```diff
@@ -0,0 +1,432 @@
+
+        let mut filtered: Vec<(usize, usize)> = Vec::new();
+        for (mut s, mut e) in ranges {
+            if self.spec.trim {
+                while s < e && bytes[s].is_ascii_whitespace() {
+                    s += 1;
+                }
+                while e > s && bytes[e - 1].is_ascii_whitespace() {
+                    e -= 1;
+                }
+            }
+            if self.spec.include_empty || e > s {
+                filtered.push((s, e));
+            }
```
(optional) we may consider putting this filter logic into a closure, e.g. `add_range`. Then in the loop above we can directly add filtered range, e.g. `add_range(prev, m.start())`. We can avoid one intermediate `Vec`.

MrAnayDongre commented:
Applied feedback: enum UPPERCASE, removed Rust-side defaults, extracted shared split helpers, simplified range logic. 
Re-tested: 143/1 + cargo tests green

georgeh0 commented:
Thanks for contributing! I'll merge it for now and take over the remaining comments.

badmonster0 commented:
hi @MrAnayDongre https://cocoindex.io/blogs/cocoindex-changelog-2025-10-19#mranaydongre latest release out and we had a section for you. thanks for your contribution!
